app/classes/query/modules/lookup_names.rb

```diff
@@ -97,59 +97,91 @@ def add_other_spellings(min_names)
         ids = min_names.map { |min_name| min_name[1] || min_name[0] }
         return [] if ids.empty?
 
-        Name.connection.select_rows(%(
-          SELECT #{minimal_name_columns} FROM names
-          WHERE COALESCE(correct_spelling_id, id) IN (#{clean_id_set(ids)})
-        ))
+        Name.
+          where((Name[:correct_spelling_id].coalesce(Name[:id])).
+                in(clean_id_set(ids).split(",").map(&:to_i))).
```

Comment:
It really makes no sense to turn ids array into a string, then parse it right back out into an array again!  The only purpose clean_id_set serves in this construction is to limit the number of ids (and maybe to return [-1] if the set is empty because maybe mysql will balk if the set is empty?), which is not at all clear from the method name.  I assume that clean_id_set should go away with everything changing to AR / Arel, right?  Maybe we can change it to leave the input as an array?  If we really want it to convert it into a comma-joined string elsewhere we can add a temporary wrapper on top of it until those uses are gone.

Reply:
Good points: 
- It's silly to turn arrays into strings, then turn them back to arrays again;
- `clean_id_set` should go away -- eventually

I left it alone here because 
- I was too lazy to look at `clean_id_set`
- the scope of this PR was just to fix some SQL injection warnings, rather than to convert all SQL to AR / Arel; and
- `clean_id_set` is used in places that are not touched by this PR.

I will revise.